x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/cluster/routing/allocation/mapper/DataTierFieldTypeTests.java

```diff
@@ -102,4 +117,18 @@ private SearchExecutionContext createContext() {
             emptyMap()
         );
     }
+
+    private SearchExecutionContext createContextWithoutSetting() {
```

Comment:
nit: maybe we could re-use the other createContext() method by adding an argument (e.g. a bool flag) to either return a context with- or without the setting?

Reply:
Thanks for the suggestion, I'll stick with this approach as I find it clearest right now. We can always refactor if these methods grow larger and there's more duplication!